### PR TITLE
Removing instance.copy() and timeout parameter from constructor.

### PR DESCRIPTION
**Pull request by sumit-ql**
This fixes issue #46.

### CODE REVIEW

**Comment by sumit-ql** [comment]
#46

**Comment by sumit-ql** [comment]
@tseaver @sduskis Please review.

File: unit_tests/test_connection.py [hunk 1]
```diff
@@ -98,28 +92,21 @@ def test_constructor_missing_instance(self):
...
 
-        self.assertEqual(get_instance_called, [timeout])
-
     def test_constructor_explicit(self):
         autoconnect = False
         table_prefix = 'table-prefix'
         table_prefix_separator = 'sep'
-        instance_copy = _Instance()
-        instance = _Instance(copies=[instance_copy])
+        instance = _Instance()
```
[the remark this answers is not in the document]
Comment (sumit-ql) [reply unit_tests/test_connection.py]: Done.

File: src/google/cloud/happybase/connection.py [hunk 2]
```diff
@@ -167,12 +150,9 @@ def __init__(self, timeout=None, autoconnect=True, table_prefix=None,
...
 
         if instance is None:
-            self._instance = _get_instance(timeout=timeout)
+            self._instance = _get_instance()
         else:
-            if timeout is not None:
-                raise ValueError('Timeout cannot be used when an existing '
-                                 'instance is passed')
-            self._instance = instance.copy()
+            self._instance = instance
```
[the remark this answers is not in the document]
Comment (sumit-ql) [reply src/google/cloud/happybase/connection.py]: updated the code.

File: unit_tests/test_pool.py [hunk 3]
```diff
@@ -46,7 +46,6 @@ def test_constructor_defaults(self):
         self.assertEqual(queue.maxsize, size)
         for connection in queue.queue:
             self.assertTrue(isinstance(connection, Connection))
-            self.assertTrue(connection._instance is instance_copy)
```
[the remark this answers is not in the document]
Comment (sumit-ql) [reply unit_tests/test_pool.py]: Removed copies from _Instance.

File: unit_tests/test_pool.py [hunk 4]
```diff
@@ -82,10 +81,8 @@ def open(self):
         instance = _Instance([instance_copy1, instance_copy2, instance_copy3])
         connection = ConnectionWithOpen(autoconnect=False, instance=instance)
         self.assertFalse(connection._open_called)
-        self.assertTrue(connection._instance is instance_copy1)
```
[the remark this answers is not in the document]
Comment (sumit-ql) [reply unit_tests/test_pool.py]: Done.

File: unit_tests/test_pool.py [hunk 5]
```diff
@@ -117,9 +113,6 @@ def mock_get_instance(timeout=None):
 
         for connection in pool._queue.queue:
             self.assertTrue(isinstance(connection, Connection))
-            # We know that the Connection() constructor will
-            # call instance.copy().
-            self.assertTrue(connection._instance is instance_copy)
```
[the remark this answers is not in the document]
Comment (sumit-ql) [reply unit_tests/test_pool.py]: Done.

File: unit_tests/test_pool.py [hunk 6]
```diff
@@ -231,14 +224,6 @@ def __init__(self, copies=()):
         # Included to support Connection.__del__
         self._client = _Client()
 
-    def copy(self):
-        if self.copies:
-            result = self.copies[0]
-            self.copies[:] = self.copies[1:]
-            return result
-
-        return self
```
[the remark this answers is not in the document]
Comment (sumit-ql) [reply unit_tests/test_pool.py]: Deleted.

**Comment by sumit-ql** [comment]
@tseaver addressed all review comments, please take a look.

File: unit_tests/test_pool.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tseaver) [reply unit_tests/test_pool.py]: In the loop over the pool connections below, add an assertion that `connection._instance is instance`.

**Comment by tseaver** [comment]
@sumit-ql I requested just the one more assertion (in `TestPool.test_constructor_defaults`).

**Pull request merged**